### PR TITLE
clusterset to excel

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

File: lib/ConditionUtils/core/Utils.py [hunk 1]
```diff
@@ -10,14 +10,17 @@
...
 
 class Utils:
     def __init__(self, config):
         self.cfg = config
         self.scratch = config['scratch']
-        self.dfu = DataFileUtil(os.environ['SDK_CALLBACK_URL'])
+        self.callback_url = os.environ['SDK_CALLBACK_URL']
+        self.dfu = DataFileUtil(self.callback_url)
         self.kbse = KBaseSearchEngine(config['search-url'])
+        self.gen_api = GenericsAPI(self.callback_url, service_ver='dev')
```
Comment (JamesJeffryes) [inline lib/ConditionUtils/core/Utils.py]: I'm not sure we want to pin to the dev version. I'd rather not specify (should get the dev if nothing else is yet available)

File: lib/ConditionUtils/core/Utils.py [hunk 2]
```diff
@@ -76,21 +79,66 @@ def file_to_condition_set(self, params):
...
+        for cluster in clusters:
+            gene_ids = cluster.get('id_to_data_position').keys()
+            gene_idx = [data_index.index(gene_id) for gene_id in gene_ids]
+
+            for idx in gene_idx:
+                cluster_name[idx] = cluster_id
+
+            cluster_id += 1
+
+        data_df['cluster'] = cluster_name
```
Comment (JamesJeffryes) [inline lib/ConditionUtils/core/Utils.py]: What if this is a cluster set based on columns? I think we need a separate set of logic and tests for that

File: lib/ConditionUtils/core/Utils.py [hunk 3]
```diff
@@ -76,21 +79,66 @@ def file_to_condition_set(self, params):
...
+
+        clusters = data.get('clusters')
+
+        data_index = data_df.index.tolist()
+        cluster_name = [None] * len(data_index)
+
+        cluster_id = 0
+        for cluster in clusters:
+            gene_ids = cluster.get('id_to_data_position').keys()
+            gene_idx = [data_index.index(gene_id) for gene_id in gene_ids]
```
Comment (JamesJeffryes) [inline lib/ConditionUtils/core/Utils.py]: I think https://pandas.pydata.org/pandas-docs/stable/generated/pandas.Index.get_loc.html would do the same thing more efficiently

**Review by JamesJeffryes — CHANGES_REQUESTED** [review]

File: lib/ConditionUtils/core/Utils.py (hunk 1 above)
Comment (Tianhao-Gu) [reply lib/ConditionUtils/core/Utils.py]: 👍

File: lib/ConditionUtils/core/Utils.py (hunk 2 above)
Comment (Tianhao-Gu) [reply lib/ConditionUtils/core/Utils.py]: Yup. Good catch.

File: lib/ConditionUtils/core/Utils.py (hunk 3 above)
Comment (Tianhao-Gu) [reply lib/ConditionUtils/core/Utils.py]: 👍

**Comment by Tianhao-Gu** [comment]
@JamesJeffryes Could you take another look? thanks!

**Review by JamesJeffryes — APPROVED** [review]
Code looks good. Local tests pass. Output looks reasonable.

**Pull request merged**